### PR TITLE
AccountCreate updates

### PR DESCRIPTION
**Pull request by jacogr**
- Add  store to CreateAccount
- DRY up, removing duplication (single responsibility in store)
- Clean up all update callbacks
- Change FirstRun to use new creation
- tests for components changed/added

PR is much bigger (in terms of LOC) than I'm typically comfortable with submitting, but at the same time it is just the single (very large) modal.

### CODE REVIEW

**Comment by jacogr** [comment]
Not comfortable for 1.5, on-ice until 1.6 - well tested, however no additional benefit apart from making this part maintainable & DRY.

File: js/src/modals/CreateAccount/createAccount.js [hunk 1]
```diff
@@ -277,7 +279,7 @@ export default class CreateAccount extends Component {
             canCreate: true
           });
 
-          this.newError(error);
+          newError(error);
```
Comment (derhuerst) [inline js/src/modals/CreateAccount/createAccount.js]: Afaict, this is a noop, as `newError` is pure.

**Comment by ngotchac** [comment]
Logs `Warning: Failed prop type: Invalid prop 'owners' supplied to 'Summary'.` as soon as I open the Create Account modal. Not sure what's going on.

When you choose something else than the default `Create new account manually` option and then go back, the default is selected and clicking `Next` brings you to the non-default option previously selected ; it's misleading.

Importing from Geth seems broken. The addresses are imported into my Address Book.

Looks good otherwise

**Comment by jacogr** [comment]
Current Geth import on master is not working. (It doesn't filter correctly, always yielding 0 accounts.) Fixing that bug to test yields exactly the same Geth import outcome - it would seem that the Name is set before the account is actually imported, i.e. it never shows up under accounts. (Once deleted from the address book it does) Should be logged as a core issue.

Other 2 issues addressed - selection based on type in store, array or array-like object in summary (with copy in attached store)

**Comment by ngotchac** [comment]
Not sure why setting `owners: nullableProptype(arrayOrObjectProptype())` in `src/views/Accounts/Summary/summary.js` has anything to do with Account Creation ?

**Comment by jacogr** [comment]
@ngotchac We need MobX 3.0.0 for `observable.shallowObject`. (Which is the correct solution as per docs). Current version doesn't have that extension on the `observer` function.

As-is the work-around does its job.

**Pull request merged**